test/test_nn.py

```diff
@@ -8843,6 +8843,15 @@ def test_batchnorm_non_contig_cpu(self):
         self.assertTrue(ref_out.is_contiguous())
         self.assertEqual(out, ref_out)
 
+        input_bf = torch.arange(24, dtype=torch.bfloat16).reshape(1, 3, 2, 4).cpu()
+        input_bf = input_bf.permute(0, 2, 1, 3)
+        input_f = input_bf.float()
+        bn_mix = torch.nn.BatchNorm2d(2).cpu().float().eval()
```

Comment:
Same as above, why cast to cpu here?
```suggestion
        bn_mix = torch.nn.BatchNorm2d(2).float().eval()
```